// @polar-sh/sdk/dist/commonjs/models/components/meter.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Meter$outboundSchema = exports.Meter$inboundSchema = exports.MeterAggregation$outboundSchema = exports.MeterAggregation$inboundSchema = exports.MeterMetadata$outboundSchema = exports.MeterMetadata$inboundSchema = void 0;
exports.meterMetadataToJSON = meterMetadataToJSON;
exports.meterMetadataFromJSON = meterMetadataFromJSON;
exports.meterAggregationToJSON = meterAggregationToJSON;
exports.meterAggregationFromJSON = meterAggregationFromJSON;
exports.meterToJSON = meterToJSON;
exports.meterFromJSON = meterFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const countaggregation_js_1 = require("./countaggregation.js");
const filter_js_1 = require("./filter.js");
const propertyaggregation_js_1 = require("./propertyaggregation.js");
const uniqueaggregation_js_1 = require("./uniqueaggregation.js");
/** @internal */
exports.MeterMetadata$inboundSchema = z.union([z.string(), z.int(), z.number(), z.boolean()]);
/** @internal */
exports.MeterMetadata$outboundSchema = z.union([z.string(), z.int(), z.number(), z.boolean()]);
function meterMetadataToJSON(meterMetadata) {
    return JSON.stringify(exports.MeterMetadata$outboundSchema.parse(meterMetadata));
}
function meterMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MeterMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MeterMetadata' from JSON`);
}
/** @internal */
exports.MeterAggregation$inboundSchema = z.union([
    z.intersection(propertyaggregation_js_1.PropertyAggregation$inboundSchema, z.object({ func: z.literal("avg") })),
    z.intersection(propertyaggregation_js_1.PropertyAggregation$inboundSchema, z.object({ func: z.literal("max") })),
    z.intersection(propertyaggregation_js_1.PropertyAggregation$inboundSchema, z.object({ func: z.literal("min") })),
    z.intersection(propertyaggregation_js_1.PropertyAggregation$inboundSchema, z.object({ func: z.literal("sum") })),
    z.intersection(uniqueaggregation_js_1.UniqueAggregation$inboundSchema, z.object({ func: z.literal("unique") })),
    z.intersection(countaggregation_js_1.CountAggregation$inboundSchema, z.object({ func: z.literal("count") })),
]);
/** @internal */
exports.MeterAggregation$outboundSchema = z.union([
    z.intersection(propertyaggregation_js_1.PropertyAggregation$outboundSchema, z.object({ func: z.literal("avg") })),
    z.intersection(propertyaggregation_js_1.PropertyAggregation$outboundSchema, z.object({ func: z.literal("max") })),
    z.intersection(propertyaggregation_js_1.PropertyAggregation$outboundSchema, z.object({ func: z.literal("min") })),
    z.intersection(propertyaggregation_js_1.PropertyAggregation$outboundSchema, z.object({ func: z.literal("sum") })),
    z.intersection(uniqueaggregation_js_1.UniqueAggregation$outboundSchema, z.object({ func: z.literal("unique") })),
    z.intersection(countaggregation_js_1.CountAggregation$outboundSchema, z.object({ func: z.literal("count") })),
]);
function meterAggregationToJSON(meterAggregation) {
    return JSON.stringify(exports.MeterAggregation$outboundSchema.parse(meterAggregation));
}
function meterAggregationFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.MeterAggregation$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MeterAggregation' from JSON`);
}
/** @internal */
exports.Meter$inboundSchema = z.pipe(z.object({
    metadata: z.record(z.string(), z.union([z.string(), z.int(), z.number(), z.boolean()])),
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    id: z.string(),
    name: z.string(),
    filter: filter_js_1.Filter$inboundSchema,
    aggregation: z.union([
        z.intersection(propertyaggregation_js_1.PropertyAggregation$inboundSchema, z.object({ func: z.literal("avg") })),
        z.intersection(propertyaggregation_js_1.PropertyAggregation$inboundSchema, z.object({ func: z.literal("max") })),
        z.intersection(propertyaggregation_js_1.PropertyAggregation$inboundSchema, z.object({ func: z.literal("min") })),
        z.intersection(propertyaggregation_js_1.PropertyAggregation$inboundSchema, z.object({ func: z.literal("sum") })),
        z.intersection(uniqueaggregation_js_1.UniqueAggregation$inboundSchema, z.object({ func: z.literal("unique") })),
        z.intersection(countaggregation_js_1.CountAggregation$inboundSchema, z.object({ func: z.literal("count") })),
    ]),
    organization_id: z.string(),
    archived_at: z.optional(z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))))),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
        "archived_at": "archivedAt",
    });
}));
/** @internal */
exports.Meter$outboundSchema = z
    .pipe(z.object({
    metadata: z.record(z.string(), z.union([z.string(), z.int(), z.number(), z.boolean()])),
    createdAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    modifiedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    id: z.string(),
    name: z.string(),
    filter: filter_js_1.Filter$outboundSchema,
    aggregation: z.union([
        z.intersection(propertyaggregation_js_1.PropertyAggregation$outboundSchema, z.object({ func: z.literal("avg") })),
        z.intersection(propertyaggregation_js_1.PropertyAggregation$outboundSchema, z.object({ func: z.literal("max") })),
        z.intersection(propertyaggregation_js_1.PropertyAggregation$outboundSchema, z.object({ func: z.literal("min") })),
        z.intersection(propertyaggregation_js_1.PropertyAggregation$outboundSchema, z.object({ func: z.literal("sum") })),
        z.intersection(uniqueaggregation_js_1.UniqueAggregation$outboundSchema, z.object({ func: z.literal("unique") })),
        z.intersection(countaggregation_js_1.CountAggregation$outboundSchema, z.object({ func: z.literal("count") })),
    ]),
    organizationId: z.string(),
    archivedAt: z.optional(z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString())))),
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        organizationId: "organization_id",
        archivedAt: "archived_at",
    });
}));
function meterToJSON(meter) {
    return JSON.stringify(exports.Meter$outboundSchema.parse(meter));
}
function meterFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Meter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Meter' from JSON`);
}
