// @polar-sh/sdk/dist/commonjs/models/components/metrics.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Metrics$inboundSchema = void 0;
exports.metricsFromJSON = metricsFromJSON;
const z = __importStar(require("zod/v4-mini"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const metric_js_1 = require("./metric.js");
/** @internal */
exports.Metrics$inboundSchema = z.pipe(z.object({
    orders: metric_js_1.Metric$inboundSchema,
    revenue: metric_js_1.Metric$inboundSchema,
    net_revenue: metric_js_1.Metric$inboundSchema,
    cumulative_revenue: metric_js_1.Metric$inboundSchema,
    net_cumulative_revenue: metric_js_1.Metric$inboundSchema,
    costs: metric_js_1.Metric$inboundSchema,
    cumulative_costs: metric_js_1.Metric$inboundSchema,
    average_order_value: metric_js_1.Metric$inboundSchema,
    net_average_order_value: metric_js_1.Metric$inboundSchema,
    average_revenue_per_user: metric_js_1.Metric$inboundSchema,
    cost_per_user: metric_js_1.Metric$inboundSchema,
    active_user_by_event: metric_js_1.Metric$inboundSchema,
    one_time_products: metric_js_1.Metric$inboundSchema,
    one_time_products_revenue: metric_js_1.Metric$inboundSchema,
    one_time_products_net_revenue: metric_js_1.Metric$inboundSchema,
    new_subscriptions: metric_js_1.Metric$inboundSchema,
    new_subscriptions_revenue: metric_js_1.Metric$inboundSchema,
    new_subscriptions_net_revenue: metric_js_1.Metric$inboundSchema,
    renewed_subscriptions: metric_js_1.Metric$inboundSchema,
    renewed_subscriptions_revenue: metric_js_1.Metric$inboundSchema,
    renewed_subscriptions_net_revenue: metric_js_1.Metric$inboundSchema,
    active_subscriptions: metric_js_1.Metric$inboundSchema,
    monthly_recurring_revenue: metric_js_1.Metric$inboundSchema,
    committed_monthly_recurring_revenue: metric_js_1.Metric$inboundSchema,
    checkouts: metric_js_1.Metric$inboundSchema,
    succeeded_checkouts: metric_js_1.Metric$inboundSchema,
    checkouts_conversion: metric_js_1.Metric$inboundSchema,
    canceled_subscriptions: metric_js_1.Metric$inboundSchema,
    canceled_subscriptions_customer_service: metric_js_1.Metric$inboundSchema,
    canceled_subscriptions_low_quality: metric_js_1.Metric$inboundSchema,
    canceled_subscriptions_missing_features: metric_js_1.Metric$inboundSchema,
    canceled_subscriptions_switched_service: metric_js_1.Metric$inboundSchema,
    canceled_subscriptions_too_complex: metric_js_1.Metric$inboundSchema,
    canceled_subscriptions_too_expensive: metric_js_1.Metric$inboundSchema,
    canceled_subscriptions_unused: metric_js_1.Metric$inboundSchema,
    canceled_subscriptions_other: metric_js_1.Metric$inboundSchema,
    gross_margin: metric_js_1.Metric$inboundSchema,
    gross_margin_percentage: metric_js_1.Metric$inboundSchema,
    cashflow: metric_js_1.Metric$inboundSchema,
}), z.transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "net_revenue": "netRevenue",
        "cumulative_revenue": "cumulativeRevenue",
        "net_cumulative_revenue": "netCumulativeRevenue",
        "cumulative_costs": "cumulativeCosts",
        "average_order_value": "averageOrderValue",
        "net_average_order_value": "netAverageOrderValue",
        "average_revenue_per_user": "averageRevenuePerUser",
        "cost_per_user": "costPerUser",
        "active_user_by_event": "activeUserByEvent",
        "one_time_products": "oneTimeProducts",
        "one_time_products_revenue": "oneTimeProductsRevenue",
        "one_time_products_net_revenue": "oneTimeProductsNetRevenue",
        "new_subscriptions": "newSubscriptions",
        "new_subscriptions_revenue": "newSubscriptionsRevenue",
        "new_subscriptions_net_revenue": "newSubscriptionsNetRevenue",
        "renewed_subscriptions": "renewedSubscriptions",
        "renewed_subscriptions_revenue": "renewedSubscriptionsRevenue",
        "renewed_subscriptions_net_revenue": "renewedSubscriptionsNetRevenue",
        "active_subscriptions": "activeSubscriptions",
        "monthly_recurring_revenue": "monthlyRecurringRevenue",
        "committed_monthly_recurring_revenue": "committedMonthlyRecurringRevenue",
        "succeeded_checkouts": "succeededCheckouts",
        "checkouts_conversion": "checkoutsConversion",
        "canceled_subscriptions": "canceledSubscriptions",
        "canceled_subscriptions_customer_service": "canceledSubscriptionsCustomerService",
        "canceled_subscriptions_low_quality": "canceledSubscriptionsLowQuality",
        "canceled_subscriptions_missing_features": "canceledSubscriptionsMissingFeatures",
        "canceled_subscriptions_switched_service": "canceledSubscriptionsSwitchedService",
        "canceled_subscriptions_too_complex": "canceledSubscriptionsTooComplex",
        "canceled_subscriptions_too_expensive": "canceledSubscriptionsTooExpensive",
        "canceled_subscriptions_unused": "canceledSubscriptionsUnused",
        "canceled_subscriptions_other": "canceledSubscriptionsOther",
        "gross_margin": "grossMargin",
        "gross_margin_percentage": "grossMarginPercentage",
    });
}));
function metricsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Metrics$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Metrics' from JSON`);
}
