// @polar-sh/sdk/dist/esm/models/operations/eventslist.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { EventSortProperty$outboundSchema, } from "../components/eventsortproperty.js";
import { EventSource$outboundSchema, } from "../components/eventsource.js";
import { ListResourceEvent$inboundSchema, } from "../components/listresourceevent.js";
import { MetadataQuery$outboundSchema, } from "../components/subscriptionslist.js";
/** @internal */
export const EventsListQueryParamOrganizationIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
export function eventsListQueryParamOrganizationIDFilterToJSON(eventsListQueryParamOrganizationIDFilter) {
    return JSON.stringify(EventsListQueryParamOrganizationIDFilter$outboundSchema.parse(eventsListQueryParamOrganizationIDFilter));
}
/** @internal */
export const EventsListQueryParamCustomerIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
export function eventsListQueryParamCustomerIDFilterToJSON(eventsListQueryParamCustomerIDFilter) {
    return JSON.stringify(EventsListQueryParamCustomerIDFilter$outboundSchema.parse(eventsListQueryParamCustomerIDFilter));
}
/** @internal */
export const QueryParamExternalCustomerIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
export function queryParamExternalCustomerIDFilterToJSON(queryParamExternalCustomerIDFilter) {
    return JSON.stringify(QueryParamExternalCustomerIDFilter$outboundSchema.parse(queryParamExternalCustomerIDFilter));
}
/** @internal */
export const NameFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
export function nameFilterToJSON(nameFilter) {
    return JSON.stringify(NameFilter$outboundSchema.parse(nameFilter));
}
/** @internal */
export const SourceFilter$outboundSchema = z.union([EventSource$outboundSchema, z.array(EventSource$outboundSchema)]);
export function sourceFilterToJSON(sourceFilter) {
    return JSON.stringify(SourceFilter$outboundSchema.parse(sourceFilter));
}
/** @internal */
export const EventsListRequest$outboundSchema = z.pipe(z.object({
    filter: z.optional(z.nullable(z.string())),
    startTimestamp: z.optional(z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString())))),
    endTimestamp: z.optional(z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString())))),
    organizationId: z.optional(z.nullable(z.union([z.string(), z.array(z.string())]))),
    customerId: z.optional(z.nullable(z.union([z.string(), z.array(z.string())]))),
    externalCustomerId: z.optional(z.nullable(z.union([z.string(), z.array(z.string())]))),
    meterId: z.optional(z.nullable(z.string())),
    name: z.optional(z.nullable(z.union([z.string(), z.array(z.string())]))),
    source: z.optional(z.nullable(z.union([
        EventSource$outboundSchema,
        z.array(EventSource$outboundSchema),
    ]))),
    query: z.optional(z.nullable(z.string())),
    parentId: z.optional(z.nullable(z.string())),
    hierarchical: z._default(z.boolean(), false),
    page: z._default(z.int(), 1),
    limit: z._default(z.int(), 10),
    sorting: z.optional(z.nullable(z.array(EventSortProperty$outboundSchema))),
    metadata: z.optional(z.nullable(z.record(z.string(), MetadataQuery$outboundSchema))),
}), z.transform((v) => {
    return remap$(v, {
        startTimestamp: "start_timestamp",
        endTimestamp: "end_timestamp",
        organizationId: "organization_id",
        customerId: "customer_id",
        externalCustomerId: "external_customer_id",
        meterId: "meter_id",
        parentId: "parent_id",
    });
}));
export function eventsListRequestToJSON(eventsListRequest) {
    return JSON.stringify(EventsListRequest$outboundSchema.parse(eventsListRequest));
}
/** @internal */
export const EventsListResponse$inboundSchema = z.pipe(z.object({
    Result: ListResourceEvent$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
}));
export function eventsListResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => EventsListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EventsListResponse' from JSON`);
}
