// @polar-sh/sdk/dist/esm/models/components/benefitcustomsubscriber.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { BenefitCustomSubscriberProperties$inboundSchema, } from "./benefitcustomsubscriberproperties.js";
import { BenefitSubscriberOrganization$inboundSchema, } from "./benefitsubscriberorganization.js";
/** @internal */
export const BenefitCustomSubscriberMetadata$inboundSchema = z.union([z.string(), z.int(), z.number(), z.boolean()]);
export function benefitCustomSubscriberMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitCustomSubscriberMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitCustomSubscriberMetadata' from JSON`);
}
/** @internal */
export const BenefitCustomSubscriber$inboundSchema = z.pipe(z.object({
    id: z.string(),
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    type: z.literal("custom"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organization_id: z.string(),
    metadata: z.record(z.string(), z.union([z.string(), z.int(), z.number(), z.boolean()])),
    organization: BenefitSubscriberOrganization$inboundSchema,
    properties: BenefitCustomSubscriberProperties$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
    });
}));
export function benefitCustomSubscriberFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitCustomSubscriber$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitCustomSubscriber' from JSON`);
}
