// @polar-sh/sdk/dist/esm/models/components/benefitdownloadablessubscriber.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { BenefitDownloadablesSubscriberProperties$inboundSchema, } from "./benefitdownloadablessubscriberproperties.js";
import { BenefitSubscriberOrganization$inboundSchema, } from "./benefitsubscriberorganization.js";
/** @internal */
export const BenefitDownloadablesSubscriberMetadata$inboundSchema = z.union([
    z.string(),
    z.int(),
    z.number(),
    z.boolean(),
]);
export function benefitDownloadablesSubscriberMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitDownloadablesSubscriberMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitDownloadablesSubscriberMetadata' from JSON`);
}
/** @internal */
export const BenefitDownloadablesSubscriber$inboundSchema = z.pipe(z.object({
    id: z.string(),
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    type: z.literal("downloadables"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organization_id: z.string(),
    metadata: z.record(z.string(), z.union([z.string(), z.int(), z.number(), z.boolean()])),
    organization: BenefitSubscriberOrganization$inboundSchema,
    properties: BenefitDownloadablesSubscriberProperties$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
    });
}));
export function benefitDownloadablesSubscriberFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitDownloadablesSubscriber$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitDownloadablesSubscriber' from JSON`);
}
