// @polar-sh/sdk/dist/esm/models/components/benefitgithubrepository.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { BenefitGitHubRepositoryProperties$inboundSchema, BenefitGitHubRepositoryProperties$outboundSchema, } from "./benefitgithubrepositoryproperties.js";
/** @internal */
export const BenefitGitHubRepositoryMetadata$inboundSchema = z.union([z.string(), z.int(), z.number(), z.boolean()]);
/** @internal */
export const BenefitGitHubRepositoryMetadata$outboundSchema = z.union([z.string(), z.int(), z.number(), z.boolean()]);
export function benefitGitHubRepositoryMetadataToJSON(benefitGitHubRepositoryMetadata) {
    return JSON.stringify(BenefitGitHubRepositoryMetadata$outboundSchema.parse(benefitGitHubRepositoryMetadata));
}
export function benefitGitHubRepositoryMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitGitHubRepositoryMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGitHubRepositoryMetadata' from JSON`);
}
/** @internal */
export const BenefitGitHubRepository$inboundSchema = z.pipe(z.object({
    id: z.string(),
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    type: z.literal("github_repository"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organization_id: z.string(),
    metadata: z.record(z.string(), z.union([z.string(), z.int(), z.number(), z.boolean()])),
    properties: BenefitGitHubRepositoryProperties$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
    });
}));
/** @internal */
export const BenefitGitHubRepository$outboundSchema = z.pipe(z.object({
    id: z.string(),
    createdAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    modifiedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    type: z.literal("github_repository"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organizationId: z.string(),
    metadata: z.record(z.string(), z.union([z.string(), z.int(), z.number(), z.boolean()])),
    properties: BenefitGitHubRepositoryProperties$outboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        organizationId: "organization_id",
    });
}));
export function benefitGitHubRepositoryToJSON(benefitGitHubRepository) {
    return JSON.stringify(BenefitGitHubRepository$outboundSchema.parse(benefitGitHubRepository));
}
export function benefitGitHubRepositoryFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BenefitGitHubRepository$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGitHubRepository' from JSON`);
}
