// @polar-sh/sdk/dist/esm/models/components/customfieldnumber.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { CustomFieldNumberProperties$inboundSchema, CustomFieldNumberProperties$outboundSchema, } from "./customfieldnumberproperties.js";
/** @internal */
export const CustomFieldNumberMetadata$inboundSchema = z.union([z.string(), z.int(), z.number(), z.boolean()]);
/** @internal */
export const CustomFieldNumberMetadata$outboundSchema = z.union([z.string(), z.int(), z.number(), z.boolean()]);
export function customFieldNumberMetadataToJSON(customFieldNumberMetadata) {
    return JSON.stringify(CustomFieldNumberMetadata$outboundSchema.parse(customFieldNumberMetadata));
}
export function customFieldNumberMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomFieldNumberMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomFieldNumberMetadata' from JSON`);
}
/** @internal */
export const CustomFieldNumber$inboundSchema = z.pipe(z.object({
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    id: z.string(),
    metadata: z.record(z.string(), z.union([z.string(), z.int(), z.number(), z.boolean()])),
    type: z.literal("number"),
    slug: z.string(),
    name: z.string(),
    organization_id: z.string(),
    properties: CustomFieldNumberProperties$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
    });
}));
/** @internal */
export const CustomFieldNumber$outboundSchema = z.pipe(z.object({
    createdAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    modifiedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    id: z.string(),
    metadata: z.record(z.string(), z.union([z.string(), z.int(), z.number(), z.boolean()])),
    type: z.literal("number"),
    slug: z.string(),
    name: z.string(),
    organizationId: z.string(),
    properties: CustomFieldNumberProperties$outboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        organizationId: "organization_id",
    });
}));
export function customFieldNumberToJSON(customFieldNumber) {
    return JSON.stringify(CustomFieldNumber$outboundSchema.parse(customFieldNumber));
}
export function customFieldNumberFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomFieldNumber$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomFieldNumber' from JSON`);
}
