// @polar-sh/sdk/dist/esm/models/components/customfielddate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { CustomFieldDateProperties$inboundSchema, CustomFieldDateProperties$outboundSchema, } from "./customfielddateproperties.js";
/** @internal */
export const CustomFieldDateMetadata$inboundSchema = z.union([z.string(), z.int(), z.number(), z.boolean()]);
/** @internal */
export const CustomFieldDateMetadata$outboundSchema = z.union([z.string(), z.int(), z.number(), z.boolean()]);
export function customFieldDateMetadataToJSON(customFieldDateMetadata) {
    return JSON.stringify(CustomFieldDateMetadata$outboundSchema.parse(customFieldDateMetadata));
}
export function customFieldDateMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomFieldDateMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomFieldDateMetadata' from JSON`);
}
/** @internal */
export const CustomFieldDate$inboundSchema = z.pipe(z.object({
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    id: z.string(),
    metadata: z.record(z.string(), z.union([z.string(), z.int(), z.number(), z.boolean()])),
    type: z.literal("date"),
    slug: z.string(),
    name: z.string(),
    organization_id: z.string(),
    properties: CustomFieldDateProperties$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
    });
}));
/** @internal */
export const CustomFieldDate$outboundSchema = z.pipe(z.object({
    createdAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    modifiedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    id: z.string(),
    metadata: z.record(z.string(), z.union([z.string(), z.int(), z.number(), z.boolean()])),
    type: z.literal("date"),
    slug: z.string(),
    name: z.string(),
    organizationId: z.string(),
    properties: CustomFieldDateProperties$outboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        organizationId: "organization_id",
    });
}));
export function customFieldDateToJSON(customFieldDate) {
    return JSON.stringify(CustomFieldDate$outboundSchema.parse(customFieldDate));
}
export function customFieldDateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomFieldDate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomFieldDate' from JSON`);
}
