// @polar-sh/sdk/dist/esm/models/components/metrics.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { Metric$inboundSchema } from "./metric.js";
/** @internal */
export const Metrics$inboundSchema = z.pipe(z.object({
    orders: Metric$inboundSchema,
    revenue: Metric$inboundSchema,
    net_revenue: Metric$inboundSchema,
    cumulative_revenue: Metric$inboundSchema,
    net_cumulative_revenue: Metric$inboundSchema,
    costs: Metric$inboundSchema,
    cumulative_costs: Metric$inboundSchema,
    average_order_value: Metric$inboundSchema,
    net_average_order_value: Metric$inboundSchema,
    average_revenue_per_user: Metric$inboundSchema,
    cost_per_user: Metric$inboundSchema,
    active_user_by_event: Metric$inboundSchema,
    one_time_products: Metric$inboundSchema,
    one_time_products_revenue: Metric$inboundSchema,
    one_time_products_net_revenue: Metric$inboundSchema,
    new_subscriptions: Metric$inboundSchema,
    new_subscriptions_revenue: Metric$inboundSchema,
    new_subscriptions_net_revenue: Metric$inboundSchema,
    renewed_subscriptions: Metric$inboundSchema,
    renewed_subscriptions_revenue: Metric$inboundSchema,
    renewed_subscriptions_net_revenue: Metric$inboundSchema,
    active_subscriptions: Metric$inboundSchema,
    monthly_recurring_revenue: Metric$inboundSchema,
    committed_monthly_recurring_revenue: Metric$inboundSchema,
    checkouts: Metric$inboundSchema,
    succeeded_checkouts: Metric$inboundSchema,
    checkouts_conversion: Metric$inboundSchema,
    canceled_subscriptions: Metric$inboundSchema,
    canceled_subscriptions_customer_service: Metric$inboundSchema,
    canceled_subscriptions_low_quality: Metric$inboundSchema,
    canceled_subscriptions_missing_features: Metric$inboundSchema,
    canceled_subscriptions_switched_service: Metric$inboundSchema,
    canceled_subscriptions_too_complex: Metric$inboundSchema,
    canceled_subscriptions_too_expensive: Metric$inboundSchema,
    canceled_subscriptions_unused: Metric$inboundSchema,
    canceled_subscriptions_other: Metric$inboundSchema,
    gross_margin: Metric$inboundSchema,
    gross_margin_percentage: Metric$inboundSchema,
    cashflow: Metric$inboundSchema,
}), z.transform((v) => {
    return remap$(v, {
        "net_revenue": "netRevenue",
        "cumulative_revenue": "cumulativeRevenue",
        "net_cumulative_revenue": "netCumulativeRevenue",
        "cumulative_costs": "cumulativeCosts",
        "average_order_value": "averageOrderValue",
        "net_average_order_value": "netAverageOrderValue",
        "average_revenue_per_user": "averageRevenuePerUser",
        "cost_per_user": "costPerUser",
        "active_user_by_event": "activeUserByEvent",
        "one_time_products": "oneTimeProducts",
        "one_time_products_revenue": "oneTimeProductsRevenue",
        "one_time_products_net_revenue": "oneTimeProductsNetRevenue",
        "new_subscriptions": "newSubscriptions",
        "new_subscriptions_revenue": "newSubscriptionsRevenue",
        "new_subscriptions_net_revenue": "newSubscriptionsNetRevenue",
        "renewed_subscriptions": "renewedSubscriptions",
        "renewed_subscriptions_revenue": "renewedSubscriptionsRevenue",
        "renewed_subscriptions_net_revenue": "renewedSubscriptionsNetRevenue",
        "active_subscriptions": "activeSubscriptions",
        "monthly_recurring_revenue": "monthlyRecurringRevenue",
        "committed_monthly_recurring_revenue": "committedMonthlyRecurringRevenue",
        "succeeded_checkouts": "succeededCheckouts",
        "checkouts_conversion": "checkoutsConversion",
        "canceled_subscriptions": "canceledSubscriptions",
        "canceled_subscriptions_customer_service": "canceledSubscriptionsCustomerService",
        "canceled_subscriptions_low_quality": "canceledSubscriptionsLowQuality",
        "canceled_subscriptions_missing_features": "canceledSubscriptionsMissingFeatures",
        "canceled_subscriptions_switched_service": "canceledSubscriptionsSwitchedService",
        "canceled_subscriptions_too_complex": "canceledSubscriptionsTooComplex",
        "canceled_subscriptions_too_expensive": "canceledSubscriptionsTooExpensive",
        "canceled_subscriptions_unused": "canceledSubscriptionsUnused",
        "canceled_subscriptions_other": "canceledSubscriptionsOther",
        "gross_margin": "grossMargin",
        "gross_margin_percentage": "grossMarginPercentage",
    });
}));
export function metricsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Metrics$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Metrics' from JSON`);
}
