// @polar-sh/sdk/dist/esm/models/components/refund.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { RefundReason$inboundSchema, RefundReason$outboundSchema, } from "./refundreason.js";
import { RefundStatus$inboundSchema, RefundStatus$outboundSchema, } from "./refundstatus.js";
/** @internal */
export const RefundMetadata$inboundSchema = z.union([z.string(), z.int(), z.number(), z.boolean()]);
/** @internal */
export const RefundMetadata$outboundSchema = z.union([z.string(), z.int(), z.number(), z.boolean()]);
export function refundMetadataToJSON(refundMetadata) {
    return JSON.stringify(RefundMetadata$outboundSchema.parse(refundMetadata));
}
export function refundMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RefundMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RefundMetadata' from JSON`);
}
/** @internal */
export const Refund$inboundSchema = z.pipe(z.object({
    created_at: z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v))),
    modified_at: z.nullable(z.pipe(z.iso.datetime({ offset: true }), z.transform(v => new Date(v)))),
    id: z.string(),
    metadata: z.record(z.string(), z.union([z.string(), z.int(), z.number(), z.boolean()])),
    status: RefundStatus$inboundSchema,
    reason: RefundReason$inboundSchema,
    amount: z.int(),
    tax_amount: z.int(),
    currency: z.string(),
    organization_id: z.string(),
    order_id: z.string(),
    subscription_id: z.nullable(z.string()),
    customer_id: z.string(),
    revoke_benefits: z.boolean(),
}), z.transform((v) => {
    return remap$(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "tax_amount": "taxAmount",
        "organization_id": "organizationId",
        "order_id": "orderId",
        "subscription_id": "subscriptionId",
        "customer_id": "customerId",
        "revoke_benefits": "revokeBenefits",
    });
}));
/** @internal */
export const Refund$outboundSchema = z
    .pipe(z.object({
    createdAt: z.pipe(z.date(), z.transform(v => v.toISOString())),
    modifiedAt: z.nullable(z.pipe(z.date(), z.transform(v => v.toISOString()))),
    id: z.string(),
    metadata: z.record(z.string(), z.union([z.string(), z.int(), z.number(), z.boolean()])),
    status: RefundStatus$outboundSchema,
    reason: RefundReason$outboundSchema,
    amount: z.int(),
    taxAmount: z.int(),
    currency: z.string(),
    organizationId: z.string(),
    orderId: z.string(),
    subscriptionId: z.nullable(z.string()),
    customerId: z.string(),
    revokeBenefits: z.boolean(),
}), z.transform((v) => {
    return remap$(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        taxAmount: "tax_amount",
        organizationId: "organization_id",
        orderId: "order_id",
        subscriptionId: "subscription_id",
        customerId: "customer_id",
        revokeBenefits: "revoke_benefits",
    });
}));
export function refundToJSON(refund) {
    return JSON.stringify(Refund$outboundSchema.parse(refund));
}
export function refundFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Refund$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Refund' from JSON`);
}
