// @polar-sh/sdk/dist/esm/models/components/subscriptioncreatecustomer.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v4-mini";
import { remap as remap$ } from "../../lib/primitives.js";
/** @internal */
export const SubscriptionCreateCustomerMetadata$outboundSchema = z.union([z.string(), z.int(), z.number(), z.boolean()]);
export function subscriptionCreateCustomerMetadataToJSON(subscriptionCreateCustomerMetadata) {
    return JSON.stringify(SubscriptionCreateCustomerMetadata$outboundSchema.parse(subscriptionCreateCustomerMetadata));
}
/** @internal */
export const SubscriptionCreateCustomer$outboundSchema = z.pipe(z.object({
    metadata: z.optional(z.record(z.string(), z.union([z.string(), z.int(), z.number(), z.boolean()]))),
    productId: z.string(),
    customerId: z.string(),
}), z.transform((v) => {
    return remap$(v, {
        productId: "product_id",
        customerId: "customer_id",
    });
}));
export function subscriptionCreateCustomerToJSON(subscriptionCreateCustomer) {
    return JSON.stringify(SubscriptionCreateCustomer$outboundSchema.parse(subscriptionCreateCustomer));
}
